src/main/java/com/synerzip/projectmanagementapp/controller/ProjectController.java

```diff
@@ -23,17 +44,41 @@
 
 	@GET
 	@Produces(MediaType.APPLICATION_JSON)
-	@Path("{projectId}")
-	public Project getProject(@PathParam("projectId") long projectId) {
-		return service.getProject(projectId);
+	@Path("/{projectId}")
+	public Response getProject(@PathParam("projectId") long projectId) {
+		Project project = service.getProject(projectId);
+		if (project == null) {
+			throw new EntityNotFoundException("record not found with id " + projectId);
+		} else {
+			return Response.ok().entity(project).build();
+		}
 	}
 
 	@GET
 	@Produces(MediaType.APPLICATION_JSON)
-	public List<Project> getProjects(
-			@DefaultValue("1") @QueryParam("start") int start,
-			@DefaultValue("5") @QueryParam("size") int size) {
-		return service.getProjects(start, size);
+	public String getProjects(@DefaultValue("0") @QueryParam("start") int start,
+			@DefaultValue("5") @QueryParam("size") int size, @DefaultValue("") @QueryParam("query") String query)
+			throws JsonGenerationException, JsonMappingException, IOException {
+		PageResult projectList = service.getProjects(start, size, query);
+		if (projectList.getData().size() == 0) {
```

Comment:
why don't throwing this error from service and handling it in exception handler/mapper. same comment is valid for other.